### PR TITLE
Bust cache when veteran has multiple phone numbers

### PR DESCRIPTION
**Pull request by leikkisa**
connects #9795

### CODE REVIEW

File: spec/feature/intake/intake_spec.rb [hunk 1]
```diff
@@ -170,6 +170,20 @@
...
+
+        safe_click ".Select"
+        fill_in "Which form are you processing?", with: Constants.INTAKE_FORM_NAMES.ramp_election
+        find("#form-select").send_keys :enter
+        safe_click ".cf-submit.usa-button"
+
+        fill_in search_bar_title, with: "12341234"
+        click_on "Search"
+
+        expect(page).to have_current_path("/intake/review_request")
```
Comment (leikkisa) [inline spec/feature/intake/intake_spec.rb]: I confirmed the behavior when bust_can_access_cache isn't called, not sure if I should formally that in a test or if the successful test is good enough.

File: app/services/external_api/bgs_service.rb [hunk 2]
```diff
@@ -187,6 +187,11 @@ def can_access?(vbms_id)
     end
   end
 
+  def bust_can_access_cache(vbms_id)
+    cache_key = "bgs_can_access_#{current_user.css_id}_#{current_user.station_id}_#{vbms_id}"
```
Comment (pkarman) [inline app/services/external_api/bgs_service.rb]: this seems ripe to put into a `user_can_access_vbms_id_cache_key` method and re-use it in `can_access?`

File: spec/feature/intake/intake_spec.rb (hunk 1 above)
Comment (pkarman) [reply spec/feature/intake/intake_spec.rb]: I think I'd want to see an explicit test for the `cache` behavior, i.e. that the key doesn't exist when the non-unique error is caught.

Comment (leikkisa) [reply spec/feature/intake/intake_spec.rb]: 👍

File: app/services/external_api/bgs_service.rb (hunk 2 above)
Comment (leikkisa) [reply app/services/external_api/bgs_service.rb]: 👍

File: spec/feature/intake/intake_spec.rb [hunk 3]
```diff
@@ -170,6 +170,24 @@
...
+        expect(Rails.cache.exist?(cache_key)).to eq(true)
```
Comment (pkarman) [inline spec/feature/intake/intake_spec.rb]: can we mock the non-unique error so we can verify that the cache key does not exist in that case?

Comment (leikkisa) [reply spec/feature/intake/intake_spec.rb]: This is on line 175 (after the non-unique error), and then the cache gets set again on trying the intake again.

Comment (pkarman) [reply spec/feature/intake/intake_spec.rb]: ah sorry I misread that originally as checking that the inittial cache state was empty.

looks great!

**Comment by Sjones352** [comment]
This is connected #9795. Once validation instructions are provided, then I will proceed with testing

**Pull request merged**